pkg/networkpolicy/controller_test.go

```diff
@@ -92,27 +101,201 @@ func newController() *networkpolicyController {
 	npaClient := npaclientfake.NewSimpleClientset()
 	npaInformerFactory := npainformers.NewSharedInformerFactory(npaClient, 0)
 
-	controller := NewController(client,
-		&knftables.Fake{},
-		informersFactory.Networking().V1().NetworkPolicies(),
-		informersFactory.Core().V1().Namespaces(),
-		informersFactory.Core().V1().Pods(),
-		informersFactory.Core().V1().Nodes(),
-		npaClient,
-		npaInformerFactory.Policy().V1alpha1().AdminNetworkPolicies(),
-		npaInformerFactory.Policy().V1alpha1().BaselineAdminNetworkPolicies(),
-		Config{
-			AdminNetworkPolicy:         true,
-			BaselineAdminNetworkPolicy: true,
+	config := Config{
+		AdminNetworkPolicy:         true,
+		BaselineAdminNetworkPolicy: true,
+	}
+
+	networkpolicyInformer := informersFactory.Networking().V1().NetworkPolicies()
+	namespaceInformer := informersFactory.Core().V1().Namespaces()
+	podInformer := informersFactory.Core().V1().Pods()
+	nodeInformer := informersFactory.Core().V1().Nodes()
+	adminNetworkPolicyInformer := npaInformerFactory.Policy().V1alpha1().AdminNetworkPolicies()
+
+	baselineAdminNetworkPolicyInformer := npaInformerFactory.Policy().V1alpha1().BaselineAdminNetworkPolicies()
+	c := &Controller{
+		podLister: podInformer.Lister(),
+		client:    client,
+		config:    config,
+		nft:       &knftables.Fake{},
+		queue:     workqueue.NewRateLimitingQueue(workqueue.DefaultControllerRateLimiter()),
+	}
+
+	err := podInformer.Informer().AddIndexers(cache.Indexers{
+		podIPIndex: func(obj interface{}) ([]string, error) {
+			pod, ok := obj.(*v1.Pod)
+			if !ok {
+				return []string{}, nil
+			}
+			// TODO check this later or it can block some traffic
+			// unrelated to the Pod
+			if pod.Spec.HostNetwork {
+				return []string{}, nil
+			}
+			result := []string{}
+			for _, ip := range pod.Status.PodIPs {
+				result = append(result, string(ip.IP))
+			}
+			return result, nil
+		},
+	})
```

Comment:
it seems crazy to duplicate all of this in the unit test; you should just have a `newControllerInternal()` that takes the knftables arg.

or call `NewController()` and just overwrite `c.nft` afterward